doc/tutorials/acrn_on_qemu.rst

```diff
@@ -0,0 +1,268 @@
+
+1. Edit your :file:`/etc/netplan/01-netcfg.yaml` file to add acrn-br0 as below,
+
+   .. code-block:: none
+
+      network:
+         version: 2
+         renderer: networkd
+         ethernets:
+            enp1s0:
+               dhcp4: no
+         bridges:
+            acrn-br0:
+               interfaces: [enp1s0]
+               dhcp4: true
+               dhcp6: no
+
+2. Apply the new network configuration by,
+
+      a. cd /etc/netplan
+      #. sudo netplan generate
+      #. sudo netplan apply
+
+3. Create a tap interface (tap0) and add the tap interface as part of the acrn-br0.
+Copy the following files to :file:`/usr/lib/system/network`,
+
+- `50-acrn.network <https://raw.githubusercontent.com/projectacrn/acrn-hypervisor/master/misc/acrnbridge/acrn.network>`__
+- `50-tap0.netdev <https://raw.githubusercontent.com/projectacrn/acrn-hypervisor/master/misc/acrnbridge/tap0.netdev>`__
+
+4. Restart ACRNSOS guest (L1 guest) to complete the setup and start with bring-up of UOS
+
+
+Bring-up UOS (L2 Guest)
+***********************
+1. Build the device-model, using make devicemodel and copy acrn-dm to ACRNSOS guest (L1 guest) directory ``/usr/bin/acrn-dm``
+
+   .. note::
+      It should be already built as part of .._Install ACRN Hypervisor.
+
+2. On the ACRNSOS guest, install shared libraries for acrn-dm (if not already installed).
+
+   .. code-block:: none
+
+      sudo apt-get install libpciaccess-dev
+
+3. Install latest `IASL tool <https://acpica.org/downloads>`_ and copy the binary to ``/usr/sbin/iasl``.
+For this setup, used IASL 20200326 version but anything after 20190215 should be good.
+
+4. Build ACRN UOS Kernel by,
+
+      a. Clone the repo, https://github.com/projectacrn/acrn-kernel.git
+      b. cd acrn-kernel
+      c. Copy kernel_config_uos to .config
+      d. make
+      e. Copy bzImage go the ACRNSOS (L1 guest). Need this to launch the UOS (L2 guest)
+
+5. Build ubuntu.img using :ref:`build-the-ubuntu-kvm-image` and copy it to the ACRNSOS (L1 Guest).
+Alternatively you can also use virt-install to create a UOS image similar to ACRNSOS as shown below,
+
+   .. code-block:: none
+
+      virt-install \
+      --name UOS \
+      --ram 2048 \
+      --disk path=/var/lib/libvirt/images/UOSUbuntu.img,size=8 \
+      --vcpus 2 \
+      --virt-type kvm \
+      --os-type linux \
+      --os-variant ubuntu18.04 \
+      --graphics none \
+      --location 'http://archive.ubuntu.com/ubuntu/dists/bionic/main/installer-amd64/' \
+      --extra-args "console=tty0 console=ttyS0,115200n8"
+
+   .. note::
+      Image at /var/lib/libvirt/images/UOSUbuntu.img is qcom2 image. Convert it to raw image using, ``qemu-img convert -f qcow2 UOSUbuntu.img -O raw UOS.img``
+
+6. Launch UOS using `launch_ubuntu.sh <https://raw.githubusercontent.com/projectacrn/acrn-hypervisor/master/doc/tutorial/launch_ubuntu.sh>`_ script. Make sure to update with your ubuntu image and rootfs
```

Comment:
If that script is in the code repo we've cloned above, it's better to use it rather than pointing a master (which will change over time)

Reply:
No this is not part of sample scripts we have in master. I can add it there if that is okay but we must ensure it is maintained.

Reply:
This would be my preference. Regardless of where it is our repo, it needs to be maintained. So from that point of view, I'd rather have it where all the others already are.

Reply:
Currently, we are supporting script files for 3 platforms in the devicemodel/samples directory. launch_ubuntu.sh doesn't fit in any of those as it is not platform-specific. So just placing this in the devicemodel/samples directory.